### DISCUSSION TITLE
Numerical artifacts when using SubdomainBoundingBoxGenerator- block-restricted variables existing outside of their blocks

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#26232] [Q&A Meshing] [answered]
Hi, I was wondering if there are any fixes to this other than meshing the interface very finely. I created a mesh with gmsh but due to the fact that I automated the process (since I need to alter the configurations slightly each time), I cannot define physical groups properly. If you've used gmsh, the "Coherence/RemoveAllDuplicates" option used to make a mesh with multiple subvolumes coherent makes it harder to define physical groups by geometry.

So I have to use MOOSE's inbuilt SubdomainBoundingBoxGenerator. In the image below, I have a very thin layer over another layer, and I used the above object to define each of them as blocks.

```
  [sample_block]
    type = SubdomainBoundingBoxGenerator
    input = sample_mesh
    block_id = 1
    top_right = '40 0 0'
    bottom_left = '-40 -40 0'
  []
  [transducer_block]
    type = SubdomainBoundingBoxGenerator
    input = sample_block
    block_id = 2	
    top_right = '40 ${transducer_thickness} 0'
    bottom_left = '-40 0 0'
  []
```

![2d_large](https://github.com/idaholab/moose/assets/52802063/c00a8a0c-57ef-4ccc-a9a2-2b2a5dcb32e3)

zoomed in, it looks like this:

![mesh_zoom](https://github.com/idaholab/moose/assets/52802063/c1768144-c792-438f-bce1-4e255e43b9ce)

The mesh is less refined further away from points of interest, and the object seems to be including elements from the substrate/lower block the less refined it as at the interface. However I am not sure why it would include those other elements at all, since the mesh was created in a way that elements terminate EXACTLY at the interface. So it should be easier for it to make a "clean break" between the upper and lower boxes. See the mesh below:

![wireframe](https://github.com/idaholab/moose/assets/52802063/cc02f712-3994-4182-bf9f-8438640aa2db)

and zoomed in, for example:

![mesh_zoomed](https://github.com/idaholab/moose/assets/52802063/a5c3e250-23b5-40f5-8891-b9afdf3e8699)

Is there anything I can do to fix this?

**Comment by GiudGiud** [comment, marked as answer]
Hello

This could also be a visualization artefact btw. 
The temp_trans variable, is it only defined in the sliver on top? Or is it defined on both blocks with a sharp transition?
What kind of finite element family and order is it?
Having a non-zero value on a node, with zeros all around, logically creates these shapes you see

Guillaume

**Reply by richmondodufisan** [reply]
Yes its only defined on the sliver on top, and its first order. And yeah I guess that makes sense, so that means the exodus file is outputting the solution at the nodes only then, and the artefacts are a linear interpolation.

**Reply by GiudGiud** [reply]
Yes. If you project using a projectionAux to a constant monomial it should look a little better
